src/main/scala/fs2/Process1.scala

```diff
@@ -45,6 +45,29 @@ object process1 {
         case chunk #: h => Pull.output(chunk) >> take(n - chunk.size.toLong).apply(h)
       }
 
+  def takeWhile[F[_], I](p: I => Boolean)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] = 
+    h => 
+      Pull.await1Option(h).flatMap { 
+        case Some(c #: h) if p(c) =>
+          Pull.output1(c) >> takeWhile(p).apply(h)
+        case _ => Pull.done
+      }
+      
+  def drop[F[_], I](n: Long)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] = 
+    h =>
+      if (n <= 0) id.apply(h)
+      else Pull.awaitLimit(if (n <= Int.MaxValue) n.toInt else Int.MaxValue)(h).flatMap {
+        case chunk #: h => drop(n - chunk.size).apply(h)
+      }
+      
+  def dropWhile[F[_], I](p: I => Boolean)(implicit F: NotNothing[F]): Handle[F, I] => Pull[F, I, Handle[F, I]] =
+    h => 
+      Pull.await1Option(h).flatMap {
```

Comment:
Even though it is prettier to write this and `takeWhile` in terms of `await1Option`, it should preserve chunkiness. In client code, you can write in whatever style you prefer, but for the FS2 library functions I'd like to preserve chunkiness whenever feasible.

If @fthomas has not done this already, I would add a method to `Chunk`:

```
trait Chunk[+A] {
  /** Return the first index whose element satisfies the predicate. 
      `c.indexOf(f).map(i => f(c(i)))` should be `Some(true)` or `None` if no item matches. */
  def indexOf(f: A => Boolean): Option[Int] = /* implement using .iterator */
```
The implement `takeWhile` and `dropWhile` in terms of this.

Other than that this looks good.

@fthomas @ShengC @djspiewak @pchlupacek I created a wiki page just to help people coordinate: https://github.com/scalaz/scalaz-stream/wiki/FS2-upgrade-wiki-page

Idea is there are a lot of functions to upgrade, and it would be nice to get some concurrency without having people accidentally duplicate work.